packages/arb-token-bridge-ui/src/components/TransferPanel/EstimatedGas.tsx

```diff
@@ -0,0 +1,110 @@
+import { InformationCircleIcon } from '@heroicons/react/24/outline'
+import { useMemo } from 'react'
+import { twMerge } from 'tailwind-merge'
+
+import { ChainLayer, useChainLayers } from '../../hooks/useChainLayers'
+import { useNetworksAndSigners } from '../../hooks/useNetworksAndSigners'
+import { useAppState } from '../../state'
+import { getNetworkName, isNetwork } from '../../util/networks'
+import { Tooltip } from '../common/Tooltip'
+import { formatAmount, formatUSD } from '../../util/NumberUtils'
+import { useNativeCurrency } from '../../hooks/useNativeCurrency'
+import { useETHPrice } from '../../hooks/useETHPrice'
+import { useGasSummaryStore } from '../../hooks/TransferPanel/useGasSummaryStore'
+
+const depositGasFeeTooltip = ({
+  l1NetworkName,
+  l2NetworkName,
+  depositToOrbit = false
+}: {
+  l1NetworkName: string
+  l2NetworkName: string
+  depositToOrbit?: boolean
+}) => ({
+  L1: `${l1NetworkName} fees go to Ethereum Validators.`,
+  L2: `${
+    depositToOrbit ? l1NetworkName : l2NetworkName
+  } fees are collected by the chain to cover costs of execution. This is an estimated fee, if the true fee is lower, you'll be refunded.`,
+  Orbit: `${l2NetworkName} fees are collected by the chain to cover costs of execution. This is an estimated fee, if the true fee is lower, you'll be refunded.`
+})
+
+export function EstimatedGas({ layer }: { layer: 'parent' | 'child' }) {
+  const {
+    app: { isDepositMode, selectedToken }
+  } = useAppState()
+  const { l1, l2 } = useNetworksAndSigners()
+  const { parentLayer, layer: childLayer } = useChainLayers()
+  const { ethToUSD } = useETHPrice()
+  const nativeCurrency = useNativeCurrency({ provider: l2.provider })
+  const parentChainNativeCurrency = useNativeCurrency({ provider: l1.provider })
+  const isParentLayer = layer === 'parent'
+  const {
+    gasSummary: { status, estimatedL1GasFees, estimatedL2GasFees }
+  } = useGasSummaryStore()
+  const l1NetworkName = getNetworkName(l1.network.id)
+  const l2NetworkName = getNetworkName(l2.network.id)
+  const isBridgingETH = selectedToken === null && !nativeCurrency.isCustom
+  const showPrice = isBridgingETH && !isNetwork(l1.network.id).isTestnet
+  const gasForLayer = isParentLayer ? parentLayer : childLayer
+  // only hide gas estimation for parent layer at withdrawal mode
+
+  const showBreakdown = isDepositMode || !isParentLayer
+
+  const estimatedGasFee = useMemo(() => {
```

Comment:
NIT: we don't need `useMemo` here, it's a number, and we don't do anything expensive

Reply:
i'd keep it rather than using IIFE as discussed